pkgs/development/libraries/azure-sdk-for-cpp/core-amqp.nix

```
@@ -0,0 +1,78 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  ninja,
+  core,
+  c-shared-utility,
+  macro-utils-c,
+  umock-c,
+  nix-update-script,
+  meta,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "azure-sdk-for-cpp-core-amqp";
+  version = "1.0.0-beta.11";
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "Azure";
+    repo = "azure-sdk-for-cpp";
+    tag = "azure-core-amqp_1.0.0-beta.11";
+    hash = "sha256-MQsz5Dmv1BwfUaN1VXMC3hPdMHihlgOBaukp5wgTNJc=";
+  };
+  sourceRoot = "source/sdk/core/azure-core-amqp";
+
+  postPatch = ''
+    sed -i '/CMAKE_CXX_STANDARD/d' CMakeLists.txt
+  '';
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
+  propagatedBuildInputs = [
+    core
+    c-shared-utility
+    macro-utils-c
+    umock-c
+  ];
```

`propagatedBuildInputs` should be avoided in most cases and should only be used to workaround build system's limitation.

A typical use case is when a package provides some CMake/pkg-config configuration which, when used by other packages, expects certain indirect dependencies to be present. Is that the reason why you are using it?

```suggestion
  buildInputs = [
    core
    c-shared-utility
    macro-utils-c
    umock-c
  ];
```

---

> Is that the reason why you are using it?

Precisely. For example `messaging-eventhubs` doesn't list any of those as direct dependencies, but fails to configure if they are not present transitively.